kdbg/dbgmainwnd.cpp

```diff
@@ -846,11 +872,23 @@ void DebuggerMainWnd::slotFileGlobalSettings()
 	m_backTimeout = prefMisc.backTimeout();
 	m_tabWidth = prefMisc.tabWidth();
 	m_sourceFilter = prefMisc.sourceFilter();
+
+	/**
+	 * Changing the m_asmGlobalFlavor doesn't trigger a reload.
+	 * It will be the default flavor the next time KDbg is started.
+	 */
+	m_asmGlobalFlavor = prefDebugger.disassemblyFlavor();
+
 	if (m_sourceFilter.isEmpty())
 	    m_sourceFilter = defaultSourceFilter;
+
 	m_headerFilter = prefMisc.headerFilter();
+
 	if (m_headerFilter.isEmpty())
 	    m_headerFilter = defaultHeaderFilter;
+
+	//If the QDialog is accepted, save at once the config
+	saveSettings(KSharedConfig::openConfig());
```

Comment:
I can use a function like the one you wrote, but why the settings shouldn't immediately saved for any interested part to "see" them? Honest question.

Reply:
You have a point here. Still, I would not want to add what is not necessary "just because we could use it some day". That is unnecessary baggage to carry around. (Others call it "over-engineering".)

Reply:
Ok, I will try to find another way. But first I must ask: If I load some binary let's say test and I set a temporary flavor let's say att at "this program", wouldn't it make sense if I load a second binary (test 2) to be able to have it's own temporary flavor (let's say Intel)? 

Or am I getting this in the wrong way?

Reply:
Yes, I think that should be possible. One of the options in *This Program* should be "default" in addition to "intel" and "att", but I think you have accounted for that already.

> Ok, I will try to find another way.

Yes, please. See, the purpose of the `KConfig` framework and `saveSettings` is to persist settings. They should not be abused as a medium to transfer pieces of information between parts of the program.